system/extensions/esp32/wifi.toit

```diff
@@ -248,15 +254,34 @@ class WifiModule implements NetworkModule:
     state := ip_events_.wait
     if (state & WIFI_IP_ASSIGNED) == 0: throw "IP_ASSIGN_FAILED"
     ip_events_.clear_state WIFI_IP_ASSIGNED
-    ip ::= wifi_get_ip_ resource_group_
+    ip ::= wifi_get_ip_ resource_group_ OWN_ADDRESS_INDEX_
     address_ = net.IpAddress ip
     logger_.info "network address dynamically assigned through dhcp" --tags={"ip": address_}
+    configure_dns_ --from_dhcp
     ip_events_.set_callback:: on_event_ it
 
   wait_for_static_ip_address_ -> none:
-    ip ::= wifi_get_ip_ resource_group_
+    ip ::= wifi_get_ip_ resource_group_ OWN_ADDRESS_INDEX_
     address_ = net.IpAddress ip
     logger_.info "network address statically assigned" --tags={"ip": address_}
+    configure_dns_ --from_dhcp=false
+
+  configure_dns_ --from_dhcp/bool -> none:
+    main_dns := wifi_get_ip_ resource_group_ MAIN_DNS_ADDRESS_INDEX_
```

Comment:
Maybe this should return null instead of NULL_IP? It would be simpler to deal with here.